src/options.js

```diff
@@ -169,24 +169,38 @@ export function dataify(data) {
 export function arrayify(values) {
   if (values == null || isArray(values)) return values;
   if (isArrowVector(values)) return maybeTypedArrowify(values);
-  switch (values.type) {
-    case "FeatureCollection":
-      return values.features;
-    case "GeometryCollection":
-      return values.geometries;
-    case "Feature":
-    case "LineString":
-    case "MultiLineString":
-    case "MultiPoint":
-    case "MultiPolygon":
-    case "Point":
-    case "Polygon":
-    case "Sphere":
-      return [values];
+  if (isGeoJSON(values)) {
+    switch (values.type) {
+      case "FeatureCollection":
+        return values.features;
+      case "GeometryCollection":
+        return values.geometries;
+      default:
+        return [values];
+    }
   }
   return Array.from(values);
 }
 
+// Duck typing test for GeoJSON
+function isGeoJSON(x) {
+  return (
+    typeof x?.type === "string" &&
+    [
+      "FeatureCollection",
+      "GeometryCollection",
+      "Feature",
+      "LineString",
+      "MultiLineString",
+      "MultiPoint",
+      "MultiPolygon",
+      "Point",
+      "Polygon",
+      "Sphere"
+    ].includes(x.type)
```

Comment:
I believe using array.includes is quite a bit slower than using a switch statement since the array must be materialized each time the test is applied, causing more garbage collection. I’d prefer to use a switch statement.